### DISCUSSION TITLE
Loading 1bit bitmap from byte array

### DISCUSSION THREAD

**Discussion by Int32Overflow** [SixLabors/ImageSharp#1253] [Q&A] [answered]
Hello,

I have a 1bit bitmap which is stored in a byte array. That means 1 byte has 8 pixels. How can I load this array as an image?

**Comment by brianpopow** [comment, marked as answer]
ImageSharp can decode 1bit bitmaps, but the bitmap needs to have a valid bitmap header (a valid file header and a valid bitmap header). This contains information on how the data has to be interpreted. One example 1bit bitmap would be [this](http://entropymine.com/jason/bmpsuite/bmpsuite/g/pal1.bmp)

If that's the case for your byte array just use:

`using (var image = Image.Load(byteArray))`

If that's not the case you would need to prepend a valid bitmap header, but i guess that's too cumbersome.

**Reply by antonfirsov** [reply]
In the case of "raw" bitmap data, it might be easier to manually widen bits to `byte`-s (`0` or `255`) and use `Image.Load<L8>(bytes)`.

**Reply by Int32Overflow** [reply]
Thanks for this great idea 👍 This solution works perfectly.

Code-Snippet:
`Image.LoadPixelData<L8>(byteData, Width, Height);`
